python/veles/util/helpers.py

```diff
@@ -21,15 +31,73 @@ def prepare_auth_key(key):
 
 def get_client_argparse():
     parser = argparse.ArgumentParser()
-    parser.add_argument('server', help='server address')
-    parser.add_argument('server_auth_key', help='hex-encoded server auth-key')
+    parser.add_argument(
+        'server_url',
+        help='server URL in form of: VELES[+SSL|+UNIX]://auth_key[:cert-'
+             'fingerprint]@([host]:port|path) '
+             'there are 3 supported schemas: VELES - TCP socket, '
+             'VELES+SSL - SSL socket '
+             'VELES+UNIX - UNIX socket (not available on Windows)')
     return parser
 
 
 def get_logging_argparse():
-    parser = argparse.ArgumentParser()
+    parser = argparse.ArgumentParser(add_help=False)
     parser.add_argument(
         '--log-level', default='INFO',
         help='set log level to one of CRITICAL, ERROR, WARNING, INFO or DEBUG',
         choices=['CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG'])
     return parser
+
+
+def parse_url(url):
+    scheme, url = url.split('://', 1)
+    scheme = UrlScheme(scheme.lower())
+    auth, loc = url.split('@', 1)
+    auth = auth.split(':', 1)
+    auth_key = auth[0]
+    if len(auth) == 1:
+        fingerprint = None
+    else:
+        fingerprint = auth[1].replace(':', '')
+    if scheme == UrlScheme.UNIX_SCHEME:
+        path = loc
+        host, port = None, None
+    else:
+        path = None
+        host, _, port = loc.rpartition(':')
+        port = int(port)
+    Url = collections.namedtuple(
+        'URL', ['scheme', 'auth_key', 'fingerprint', 'host', 'port', 'path'])
+    return Url(scheme, auth_key, fingerprint, host, port, path)
+
+
+def generate_ssl_cert(cert_path, key_path):
+    key = crypto.PKey()
+    key.generate_key(crypto.TYPE_RSA, 2048)
+
+    cert = crypto.X509()
+    cert.get_subject().C = "PL"
+    cert.get_subject().O = "Codilime"
+    cert.get_subject().OU = "Codisec"
+    cert.set_serial_number(1000)
+    cert.gmtime_adj_notBefore(0)
+    cert.gmtime_adj_notAfter(10 * 365 * 24 * 60 * 60)
+    cert.set_issuer(cert.get_subject())
+    cert.set_pubkey(key)
+    cert.sign(key, 'sha1')
+
+    with open(cert_path, 'w') as f:
+        f.write(crypto.dump_certificate(crypto.FILETYPE_PEM, cert).decode())
+    with open(key_path, 'w') as f:
```

Comment:
what is default permission of this file? if it is readable by all, maybe it is worth to change that?